### ISSUE TITLE
unable to get timestamp from EKS logs

### ISSUE THREAD

**Issue by GAHila** [fluent-plugins-nursery/fluent-plugin-cloudwatch-logs#147]
When we try to download EKS logs (which except kube-apiserver-audit are not json logs) first of all with the default format which seems to be json we get an error.

So I changed the format to none.

This gets only the message that is provided from AWS (not the timestamp nor ingestionTime) which looks like:

`E0418 14:07:54.344381       1 resource_quota_controller.go:430] unable to retrieve the complete list of server APIs: custom.metrics.k8s.io/v1beta1: the server could not find the requested resource`

The message itself has only time - not a date so we cannot get the timestamp for the log from message field.

AWS cloudwatch logs when pulled through CLI look like:

```
        {
            "ingestionTime": 1555596478973, 
            "timestamp": 1555596474000, 
            "message": "E0418 14:07:54.344381       1 resource_quota_controller.go:430] unable to retrieve the complete list of server APIs: custom.metrics.k8s.io/v1beta1: the server could not find the requested resource"
        }, 
        {
            "ingestionTime": 1555596489067, 
            "timestamp": 1555596475000, 
            "message": "E0418 14:07:55.147299       1 memcache.go:147] couldn't get resource list for custom.metrics.k8s.io/v1beta1: the server could not find the requested resource"
        }
```

Yet when we load the logs (with format none or a regex that includes everything format /^(?<log_message>.*)/ ) the timestamp showing up in Elasticsearch is the timestamp that the log was loaded into Elasticsearch rather than the timestamp of when the log is created. That makes logs unsearchable as if for some reason we are left behind hours or even days would be impossible to search by log timestamp.

Is there any way we can get the timestamp field from AWS log itself? 

Here is the current configuration:

```
    <source>
      @type cloudwatch_logs
      tag eks-controller
      region us-east-1 
      aws_use_sts true
      aws_sts_role_arn arn:aws:iam::33434342424:role/cloudwatch-fluentd-role
      log_group_name /aws/eks/k8s/cluster
      log_stream_name kube-controller-manager
      state_file /cloudwatchlog-state/eks-controller_stream.in.state
      use_log_stream_name_prefix true
      format /^(?<log_message>.*)/
      #format none
    </source>

    <match eks-controller>
      @type copy
      <store>
        @type elasticsearch
        @log_level info
        include_tag_key true
        scheme http
        host elasticsearch-logging
        port 9200
        logstash_format true
        logstash_prefix eks-controller-k8s
        logstash_prefix_separator _
        <buffer>
          @type file
          path /cloudwatchlog-state/eks-controller.k8s.dev.buffer
          flush_mode interval
          retry_type exponential_backoff
          flush_thread_count 2
          flush_interval 5s
          retry_forever
          retry_max_interval 30
          chunk_limit_size 2M
          queue_limit_length 8
          overflow_action block
        </buffer>
      </store>
    </match>
```

**Comment by GAHila** [comment]
By the way I tried both json_handler json and yajl, they both fail (with setting format json or not as it seems to be default).

JSON handler:

```
2019-04-18 20:03:43 +0000 [warn]: thread exited by unexpected error plugin=Fluent::Plugin::CloudwatchLogsInput title=:in_cloudwatch_logs_runner error_class=JSON::ParserError error="822: unexpected token at 'W0418 20:03:13.440583       1 garbagecollector.go:647] failed to discover some groups: map[custom.metrics.k8s.io/v1beta1:the server could not find the requested resource]'"
2019-04-18 20:03:44 +0000 [warn]: thread exited by unexpected error plugin=Fluent::Plugin::CloudwatchLogsInput title=:in_cloudwatch_logs_runner error_class=Yajl::ParseError error="lexical error: invalid char in json text.\n                                       I0418 20:03:10.080617       1 g\n                     (right here) ------^\n"
2019-04-18 20:03:44 +0000 [error]: unexpected error error_class=Yajl::ParseError error="lexical error: invalid char in json text.\n                                       I0418 20:03:10.080617       1 g\n                     (right here) ------^\n"
  2019-04-18 20:03:44 +0000 [error]: /var/lib/gems/2.3.0/gems/yajl-ruby-1.4.1/lib/yajl.rb:44:in `parse'
  2019-04-18 20:03:44 +0000 [error]: /var/lib/gems/2.3.0/gems/yajl-ruby-1.4.1/lib/yajl.rb:44:in `parse'
  2019-04-18 20:03:44 +0000 [error]: /var/lib/gems/2.3.0/gems/yajl-ruby-1.4.1/lib/yajl.rb:18:in `load'
  2019-04-18 20:03:44 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:144:in `emit'
  2019-04-18 20:03:44 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:119:in `block (2 levels) in run'
  2019-04-18 20:03:44 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:118:in `each'
  2019-04-18 20:03:44 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:118:in `block in run'
  2019-04-18 20:03:44 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:115:in `each'
  2019-04-18 20:03:44 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:115:in `run'
  2019-04-18 20:03:44 +0000 [error]: /var/lib/gems/2.3.0/gems/fluentd-1.3.3/lib/fluent/plugin_helper/thread.rb:78:in `block in thread_create'
2019-04-18 20:03:44 +0000 [error]: unexpected error error_class=Yajl::ParseError error="lexical error: invalid char in json text.\n                                       I0418 20:03:10.080617       1 g\n                     (right here) ------^\n"
  2019-04-18 20:03:44 +0000 [error]: suppressed same stacktrace
```

Yajl handler:

```
2019-04-18 20:12:27 +0000 [warn]: thread exited by unexpected error plugin=Fluent::Plugin::CloudwatchLogsInput title=:in_cloudwatch_logs_runner error_class=Yajl::ParseError error="lexical error: invalid char in json text.\n                                       I0418 20:11:49.384889       1 g\n                     (right here) ------^\n"
2019-04-18 20:12:27 +0000 [error]: unexpected error error_class=Yajl::ParseError error="lexical error: invalid char in json text.\n                                       I0418 20:11:49.384889       1 g\n                     (right here) ------^\n"
  2019-04-18 20:12:27 +0000 [error]: /var/lib/gems/2.3.0/gems/yajl-ruby-1.4.1/lib/yajl.rb:44:in `parse'
  2019-04-18 20:12:27 +0000 [error]: /var/lib/gems/2.3.0/gems/yajl-ruby-1.4.1/lib/yajl.rb:44:in `parse'
  2019-04-18 20:12:27 +0000 [error]: /var/lib/gems/2.3.0/gems/yajl-ruby-1.4.1/lib/yajl.rb:18:in `load'
  2019-04-18 20:12:27 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:144:in `emit'
  2019-04-18 20:12:27 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:119:in `block (2 levels) in run'
  2019-04-18 20:12:27 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:118:in `each'
  2019-04-18 20:12:27 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:118:in `block in run'
  2019-04-18 20:12:27 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:115:in `each'
  2019-04-18 20:12:27 +0000 [error]: /var/lib/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.7.3/lib/fluent/plugin/in_cloudwatch_logs.rb:115:in `run'
  2019-04-18 20:12:27 +0000 [error]: /var/lib/gems/2.3.0/gems/fluentd-1.3.3/lib/fluent/plugin_helper/thread.rb:78:in `block in thread_create'
2019-04-18 20:12:27 +0000 [error]: unexpected error error_class=Yajl::ParseError error="lexical error: invalid char in json text.\n                                       I0418 20:11:49.384889       1 g\n                     (right here) ------^\n"
  2019-04-18 20:12:27 +0000 [error]: suppressed same stacktrace
```

**Comment by GAHila** [comment]
I created a pull request for this to get timestamp for non json logs from aws event as well.
See here:

https://github.com/fluent-plugins-nursery/fluent-plugin-cloudwatch-logs/pull/148

Please take a look

**Comment by GAHila** [comment]
> I created a pull request for this to get timestamp for non json logs from aws event as well.
> See here:
> 
> #148
> 
> Please take a look @cosmo0920 @okkez @yuki24

Code is merged into master. Waiting on new release version.
